--- CMakeLists.txt.orig	2015-08-13 13:40:39.265206289 +0900
+++ CMakeLists.txt	2015-08-13 13:41:26.770487838 +0900
@@ -19,7 +19,7 @@
 	set( LIBRARY_OUTPUT_DIRECTORY ./lib )
 ENDIF()
 IF(NOT DEFINED INCLUDE_OUTPUT_DIRECTORY)
-	set( INCLUDE_OUTPUT_DIRECTORY ./include )
+	set( INCLUDE_OUTPUT_DIRECTORY ./include/genfit2 )
 ENDIF()
 
 # define path to macros and settings
@@ -344,7 +344,7 @@
 		${PROJECT_NAME}
 		${ROOT_LIBS}  
   		${RAVE_LDFLAGS_STR}
-  		-I${RAVE_INCLUDE_DIRS_I}
+#  		-I${RAVE_INCLUDE_DIRS_I}
   	)
 else()
 	TARGET_LINK_LIBRARIES(${PROJECT_NAME} ${ROOT_LIBS})
