diff --git src/ROOT/smatrix/Math/MConfig.h src/ROOT/smatrix/Math/MConfig.h
index d0b3786..76ad3c4 100644
--- src/ROOT/smatrix/Math/MConfig.h
+++ src/ROOT/smatrix/Math/MConfig.h
@@ -2,7 +2,7 @@
 // Authors: T. Glebe, L. Moneta    2005  
 
 #ifndef ROOT_Math_MConfig_
-#define ROOT_Math_MConfig
+#define ROOT_Math_MConfig_
 
 // for alpha streams 
 #if defined(__alpha) && !defined(linux)
diff --git src/RecoVertex/GaussianSumVertexFit/src/GsfVertexWeightCalculator.cc src/RecoVertex/GaussianSumVertexFit/src/GsfVertexWeightCalculator.cc
index f5e5793..80e7fc8 100644
--- src/RecoVertex/GaussianSumVertexFit/src/GsfVertexWeightCalculator.cc
+++ src/RecoVertex/GaussianSumVertexFit/src/GsfVertexWeightCalculator.cc
@@ -56,7 +56,7 @@ double GsfVertexWeightCalculator::calculate(const  VertexState & oldVertex,
   double weight = pow(2. * M_PI, -0.5 * 5) * sqrt(1./sigmaDet) * exp(-0.5 * chi);
   // edm::LogError("GsfVertexWeightCalculator") << "w=" << weight << " chi=" << chi << " sigmaDet=" << sigmaDet;
 
-  if (isnan(weight) || sigmaDet<=0.) {
+  if (std::isnan(weight) || sigmaDet<=0.) {
     edm::LogWarning("GsfVertexWeightCalculator") << "Weight is NaN";
     return 0.;
     // return -1.;
